pybossa/dashboard.py

```diff
@@ -0,0 +1,408 @@
+# -*- coding: utf8 -*-
+# This file is part of PyBossa.
+#
+# Copyright (C) 2015 SF Isle of Man Limited
+#
+# PyBossa is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PyBossa is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with PyBossa.  If not, see <http://www.gnu.org/licenses/>.
+"""Dashboard Jobs module for running background tasks in PyBossa server."""
+from sqlalchemy import text
+from pybossa.core import db
+from datetime import datetime
+
+MINUTE = 60
+
+
+def get_dashboard_jobs():  # pragma: no cover
+    """Return dashboard jobs."""
+    yield dict(name=dashboard_active_users_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_active_anon_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_new_projects_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_update_projects_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_new_tasks_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_new_task_runs_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_new_users_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+    yield dict(name=dashboard_returning_users_week, args=[], kwargs={},
+               timeout=(10 * MINUTE), queue='low')
+
+
+def dashboard_active_users_week():
```

Comment:
That minor change is to know where it fails. If the names are the same, it will be a bit more difficult :-)